test/extended/util/client.go

```diff
@@ -702,18 +703,54 @@ func (c *CLI) AddResourceToDelete(resource schema.GroupVersionResource, metadata
 }
 
 func (c *CLI) CreateUser(prefix string) *userv1.User {
-	user, err := c.AdminUserClient().UserV1().Users().Create(context.Background(), &userv1.User{
-		ObjectMeta: metav1.ObjectMeta{GenerateName: prefix + c.Namespace()},
-	}, metav1.CreateOptions{})
-	if err != nil {
-		FatalErr(err)
-	}
-	c.AddResourceToDelete(userv1.GroupVersion.WithResource("users"), user)
+	var (
+		user *userv1.User
+	)
+	withRetryOnTemporaryErr(context.TODO(), func() error {
+		var err error
+		user, err = c.AdminUserClient().UserV1().Users().Create(context.Background(), &userv1.User{
+			ObjectMeta: metav1.ObjectMeta{GenerateName: prefix + c.Namespace()},
+		}, metav1.CreateOptions{})
+		return err
+	})
 
+	c.AddResourceToDelete(userv1.GroupVersion.WithResource("users"), user)
 	return user
 }
 
+func withRetryOnTemporaryErr(ctx context.Context, fn func() error) {
+	retriableAPIErrors := func(err error) bool {
+		return apierrors.IsServiceUnavailable(err) || // in case API server is rolling out
+			apierrors.IsServerTimeout(err) ||
+			apierrors.IsUnexpectedServerError(err) ||
+			netutil.IsConnectionRefused(err) ||
+			netutil.IsConnectionReset(err) ||
+			netutil.IsNoRoutesError(err) ||
+			netutil.IsProbableEOF(err)
+	}
+	if err := kretry.OnError(kretry.DefaultRetry, retriableAPIErrors, func() error {
```

Comment:
isn't there a retry func with context already?

Reply:
in our library-go, wanted to avoid unnecessary import

Reply:
don't we have library-go here?